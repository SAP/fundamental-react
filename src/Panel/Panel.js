import PropTypes from 'prop-types';
import React from 'react';

// ------------------------------------------- Panel ------------------------------------------
export const Panel = props => {
    const { colSpan, children, className, ...rest } = props;
    return <div className={`fd-panel${colSpan ? ' fd-has-grid-column-span-' + colSpan : ''}${className ? ' ' + className : ''}`} {...rest}>{children}</div>;
};

Panel.propTypes = {
    className: PropTypes.string,
    colSpan: PropTypes.number
};

Panel.defaultProps = {
    colSpan: null
};

// ------------------------------------------- Panel Grid ------------------------------------------
export const PanelGrid = props => {
    const { nogap, cols, children, className, ...rest } = props;
    return (
        <div
            className={`fd-panel-grid${nogap ? ' fd-panel-grid--nogap' : ''}${
                cols ? ' fd-panel-grid--' + cols + 'col' : ''
            }${className ? ' ' + className : ''}`} {...rest}>
            {children}
        </div>
    );
};

PanelGrid.propTypes = {
    className: PropTypes.string,
    cols: PropTypes.number,
    nogap: PropTypes.bool
};

PanelGrid.defaultProps = {
    nogap: false,
    cols: null
};

// ------------------------------------------- Panel Body ------------------------------------------
export const PanelBody = props => {
    const { children, className, ...rest } = props;
    return <div className={`fd-panel__body${className ? ' ' + className : ''}`} {...rest}>{children}</div>;
};

PanelBody.propTypes = {
    className: PropTypes.string
};

// ------------------------------------------- Panel Header ------------------------------------------
export const PanelHeader = props => {
    const { children, className, ...rest } = props;
    return <div className={`fd-panel__header${className ? ' ' + className : ''}`} {...rest}>{children}</div>;
};

PanelHeader.propTypes = {
    className: PropTypes.string
};

// ------------------------------------------- Panel Head ------------------------------------------
export const PanelHead = props => {
    const { title, description, className, ...rest } = props;
    return (
        <div className={`fd-panel__head${className ? ' ' + className : ''}`} {...rest}>
            {title ? <h1 className='fd-panel__title'>{title}</h1> : null}
            {description ? <p className='fd-panel__description'>{description}</p> : null}
        </div>
    );
};

PanelHead.propTypes = {
    className: PropTypes.string
};

// ------------------------------------------- Panel Actions ------------------------------------------
export const PanelActions = props => {
    const { children, className, ...rest } = props;
    return <div className={`fd-panel__actions${className ? ' ' + className : ''}`} {...rest}>{children}</div>;
};

PanelActions.propTypes = {
    className: PropTypes.string
};

// ------------------------------------------- Panel Filters ------------------------------------------
export const PanelFilters = props => {
    const { children, className, ...rest } = props;
    return (
        <div className={`fd-panel__filters${className ? ' ' + className : ''}`} {...rest}>
            {children}
        </div>
    );
};

PanelFilters.propTypes = {
    className: PropTypes.string
};


// ------------------------------------------- Panel Content ------------------------------------------
export const PanelContent = props => {
    const { children, className, ...rest } = props;
    return <div className={`fd-panel__content${className ? ' ' + className : ''}`} {...rest}>{children}</div>;
};

PanelContent.propTypes = {
    className: PropTypes.string
};

// ------------------------------------------- Panel Footer ------------------------------------------
export const PanelFooter = props => {
    const { children, className, ...rest } = props;
    return <div className={`fd-panel__footer${className ? ' ' + className : ''}`} {...rest}>{children}</div>;
};

PanelFooter.propTypes = {
    className: PropTypes.string
};
