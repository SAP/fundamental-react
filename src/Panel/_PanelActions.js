import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const PanelActions = props => {
    const { children, className, ...rest } = props;

    const panelActionsClasses = classnames(
        'fd-panel__actions',
        className
    );

    return <div {...rest} className={panelActionsClasses}>{children}</div>;
};

PanelActions.displayName = 'Panel.Actions';

PanelActions.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default PanelActions;
