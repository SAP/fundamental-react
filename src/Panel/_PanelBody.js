import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const PanelBody = props => {
    const { children, className, ...rest } = props;

    const panelBodyClasses = classnames(
        'fd-panel__body',
        className
    );

    return <div {...rest} className={panelBodyClasses}>{children}</div>;
};

PanelBody.displayName = 'Panel.Body';

PanelBody.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default PanelBody;
