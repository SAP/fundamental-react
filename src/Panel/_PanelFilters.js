import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const PanelFilters = props => {
    const { children, className, ...rest } = props;

    const panelFiltersClasses = classnames(
        'fd-panel__filters',
        className
    );

    return (
        <div {...rest} className={panelFiltersClasses}>
            {children}
        </div>
    );
};

PanelFilters.displayName = 'Panel.Filters';

PanelFilters.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default PanelFilters;
