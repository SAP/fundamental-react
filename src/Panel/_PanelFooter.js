import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const PanelFooter = props => {
    const { children, className, ...rest } = props;

    const panelFooterClasses = classnames(
        'fd-panel__footer',
        className
    );

    return <div {...rest} className={panelFooterClasses}>{children}</div>;
};

PanelFooter.displayName = 'Panel.Footer';

PanelFooter.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default PanelFooter;
