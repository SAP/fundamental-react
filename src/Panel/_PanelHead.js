import classnames from 'classnames';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import PropTypes from 'prop-types';
import React from 'react';

const PanelHead = props => {
    const { title, description, className, headingLevel, ...rest } = props;

    const panelHeadClasses = classnames(
        'fd-panel__head',
        className
    );

    const HeadingTag = `h${headingLevel}`;

    return (
        <div {...rest} className={panelHeadClasses}>
            {title ? <HeadingTag className='fd-panel__title'>{title}</HeadingTag> : null}
            {description ? <p className='fd-panel__description'>{description}</p> : null}
        </div>
    );
};

PanelHead.displayName = 'Panel.Head';

PanelHead.propTypes = {
    className: PropTypes.string,
    description: PropTypes.string,
    headingLevel: CustomPropTypes.range(2, 6),
    title: PropTypes.string
};

PanelHead.defaultProps = {
    headingLevel: 3
};

PanelHead.propDescriptions = {
    description: 'Localized text for the description of the panel.',
    title: 'Localized text for the title of the panel.'
};

export default PanelHead;
