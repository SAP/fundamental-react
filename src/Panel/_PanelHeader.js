import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const PanelHeader = props => {
    const { children, className, ...rest } = props;

    const panelHeaderClasses = classnames(
        'fd-panel__header',
        className
    );

    return <div {...rest} className={panelHeaderClasses}>{children}</div>;
};

PanelHeader.displayName = 'Panel.Header';

PanelHeader.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default PanelHeader;
