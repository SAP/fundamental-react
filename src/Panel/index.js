export { default as Panel } from './Panel.js';
export { default as PanelGrid } from './PanelGrid.js';
