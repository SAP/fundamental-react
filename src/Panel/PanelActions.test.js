import { mount } from 'enzyme';
import Panel from './Panel';
import React from 'react';

describe('<PanelActions />', () => {

    describe('Prop spreading', () => {
        test('should allow props to be spread to the PanelActions component', () => {
            const element = mount(<Panel.Actions data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
