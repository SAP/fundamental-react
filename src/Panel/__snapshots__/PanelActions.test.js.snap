// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelActions /> create panel actions 1`] = `
<div
  className="fd-panel__actions"
>
  <button
    className="fd-button fd-button--compact sap-icon--add"
    type="button"
  >
    Add New Button
  </button>
</div>
`;
